src/ome_zarr_models/__init__.py

```diff
@@ -1,6 +1,43 @@
 from importlib.metadata import PackageNotFoundError, version
 
+import zarr
+
+import ome_zarr_models.v04.hcs
+import ome_zarr_models.v04.image
+import ome_zarr_models.v04.image_label
+import ome_zarr_models.v04.labels
+import ome_zarr_models.v04.well
+from ome_zarr_models.base import BaseGroup
+from ome_zarr_models.v04.base import BaseGroupv04
+
 try:
     __version__ = version("ome_zarr_models")
 except PackageNotFoundError:  # pragma: no cover
     __version__ = "uninstalled"
+
+
+_V04_groups: list[type[BaseGroupv04]] = [
+    ome_zarr_models.v04.hcs.HCS,
+    ome_zarr_models.v04.image_label.ImageLabel,
+    ome_zarr_models.v04.image.Image,
+    ome_zarr_models.v04.labels.Labels,
+    ome_zarr_models.v04.well.Well,
+]
+
+
+def load_ome_zarr_group(group: zarr.Group) -> BaseGroup:
```

Comment:
I think we can indulge in a shorter function name -- how about `read`

Reply:
👍 for shorter name - perhaps `open()` to copy the top level methods in `zarr-python`?

Reply:
I went with `open_ome_zarr` (since `open` is a builtin)

Reply:
`open` is a bit ambiguous about what comes out (is the result mutable? does it require a file handle?) which is why I think `open` was not a great choice for `zarr-python`. If the thing returned by this function is immutable, then i'd pick a name that emphasizes this fact